### PR TITLE
VACMS-14116 Improve Events test coverage

### PR DESCRIPTION
**Pull request by randimays**
## Summary
Improve Events unit test coverage.

Includes:
- Removes unnecessary unit test for constants file
- Some refactor / cleanup to increase readability
- Broke out several components from `Search/index.js` to improve testability and readability

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/14116

## Testing done
Unit tests to cover functionality.

|         | Statements | Branches | Functions | Lines |
| ------- | ------ | ----- | --- | --- |
| `main`  |    63%    | 57%      | 68%  |63% |
| this branch |     71% (+8%)   | 76% (+19%)      |70% (+2%) |71% (+8%) | 

### Before (`main`)
Screenshot 2023-12-21 at 5 16 21 PM

### After (this branch)
Screenshot 2023-12-21 at 5 15 36 PM

## Acceptance criteria
- [x] tests confirm that duplicate events are not exposed to the FE
- [x] tests confirm that recurrences get their own "event" object in the event listing
- [x] tests confirm that online events display correctly ("where" should not have an address, but display "This is an online event".
- [x] Active events - See https://github.com/department-of-veterans-affairs/va.gov-cms/issues/10307
- [ ] Test coverage should be at least 80%

### CODE REVIEW

File: src/applications/static-pages/events/helpers/index.js [hunk 1]
```diff
@@ -390,12 +387,3 @@ export const deriveFilteredEvents = ({
     endsAtUnix,
   });
 };
-
-export const getDisabledEndDateOptions = (options, limit) => {
```
Comment (randimays) [inline src/applications/static-pages/events/helpers/index.js:394]: This function was unused

File: src/applications/static-pages/events/helpers/index.js [hunk 2]
```diff
@@ -55,12 +55,12 @@ export const deriveMostRecentDate = (
   return futureDates[0];
 };
 
-function keepUniqueEventsFromList(allEvents, event) {
-  if (!allEvents) {
+export const addUniqueEventsToList = (allEvents, event) => {
```
Comment (randimays) [inline src/applications/static-pages/events/helpers/index.js:58]: Not sure why this was named `keepUniqueEventsFromList` previously, but its new name is more appropriate for its function.

File: src/applications/static-pages/events/components/Search/index.js [hunk 3]
```diff
@@ -1,7 +1,9 @@
 import React, { useEffect, useState } from 'react';
```
Comment (randimays) [inline src/applications/static-pages/events/components/Search/index.js:1]: Most changes took place in this file, where I extracted 3 components into separate files. Nothing at all should've changed related to functionality.

**Review by eselkin — APPROVED** [review]
Broken out components look good. Testing is good. I always pick up a few pointers in testing when I look at your tests.

**Review by chriskim2311 — APPROVED** [review]
LGTM, Nice breaking out the 3 components. Great work!

**Pull request merged**